lib/protocol/encryption.go

```diff
@@ -590,3 +595,27 @@ func isEncryptedParentFromComponents(pathComponents []string) bool {
 	}
 	return true
 }
+
+type folderKeyRegistry struct {
+	keys map[string]*[keySize]byte // folder ID -> key
+	mut  sync.RWMutex
+}
+
+func newFolderKeyRegistry() *folderKeyRegistry {
+	return &folderKeyRegistry{
+		keys: make(map[string]*[keySize]byte),
+	}
+}
+
+func (r *folderKeyRegistry) get(folder string) (*[keySize]byte, bool) {
+	r.mut.RLock()
+	key, ok := r.keys[folder]
+	r.mut.RUnlock()
+	return key, ok
+}
+
+func (r *folderKeyRegistry) reset(passwords map[string]string) {
```

Comment:
`setPasswords`? `reset` sounds like remove all passwords to me

Reply:
I was just wondering whether one might misunderstand the end-result: It is exactly `passwords` but it could also be the previous state with the elements of `passwords` applied on top of it (i.e. missing passwords remaining). `reset` feels a bit more clear in that respect than just `set`. Maybe `replace` instead?

Reply:
I think `set` implies replace, but I'm fine with `replace` as well.